### PR TITLE
engine: Make ReadOnly a toggleable atomic.Bool.

### PR DESCRIPTION
**Pull request by reltuk**

### CODE REVIEW

File: engine.go [hunk 1]
```diff
@@ -635,19 +637,15 @@ func (e *Engine) WithBackgroundThreads(b *sql.BackgroundThreads) *Engine {
...
-			return sql.ErrDatabaseWriteLocked.New()
-		}
-	}
-	switch node.(type) {
-	case
-		*plan.DeleteFrom, *plan.InsertInto, *plan.Update, *plan.LockTables, *plan.UnlockTables:
-		if e.IsReadOnly {
+	nodeIsReadOnly := plan.IsReadOnly(node)
+	if !nodeIsReadOnly {
+		if e.IsReadOnly() {
```
Comment (max-hoffman) [inline engine.go:648]: would it be incorrect to invert this check? I'd think e.IsReadyOnly() is less expensive

**Review by max-hoffman — APPROVED** [review]
lgtm, one perf Q

File: engine.go (hunk 1 above)
Comment (reltuk) [reply engine.go]: Great point. Just pushed a fix. PTAL.

**Review by max-hoffman — APPROVED** [review]
still lgtm, i was meaning to switch that check anyways because this rule shows up on profiles, thanks for saving me the time

**Pull request merged**